drv/stm32h7-sprot-server/src/main.rs

```diff
@@ -404,32 +422,35 @@ impl<S: SpiServer> Io<S> {
                                 ringbuf_entry!(Trace::ErrRespNoPayload);
                                 continue;
                             }
-                            errcode =
-                                SprotError::from(rxmsg.payload_error_byte());
-                            ringbuf_entry!(Trace::SprotError(errcode));
-                            if is_recoverable_error(errcode) {
-                                // TODO: There are rare cases where
-                                // the RoT dose not receive
-                                // a 0x01 as the first byte in a message.
-                                // See issue #929.
+                            let err =
+                                SprotError::from_u8(rxmsg.payload_error_byte())
+                                    .unwrap_or(SprotError::Unknown);
+                            errcode = Some(err);
+                            ringbuf_entry!(Trace::SprotError(err));
+                            if is_recoverable_error(err) {
```

Comment:
`is_recoverable()` could be a method on `err`, especially if we have a new `SpSprotErr` type (see above).